lib/elixir/lib/dynamic_supervisor.ex

```diff
@@ -1136,15 +1136,6 @@ defmodule DynamicSupervisor do
     ]
   end
 
-  @impl true
-  def format_status(:terminate, [_pdict, state]) do
-    state
-  end
-
-  def format_status(_, [_pdict, %{mod: mod} = state]) do
-    [data: [{~c"State", state}], supervisor: [{~c"Callback", mod}]]
```

Comment:
I don't think we can reproduce this behavior with `format_status/1`, which breaks a test - should I just keep this one around for now?

Reply:
Or we can just remove these [tests](https://github.com/elixir-lang/elixir/blob/v1.16.2/lib/elixir/test/elixir/dynamic_supervisor_test.exs#L144-L150), or implement them to use `:mod`?

If `:supervisor.get_callback_module` is used somewhere which matters, should erlang add a key to the `format_status` map?

Reply:
@sabiwara we should mirror whatever the Erlang `supervisor` is doing. This was most likely just borrowed from them.

Reply:
OK nice, seems they just removed it.
https://github.com/erlang/otp/commit/67513dcdd3823fa75dbd6b8a255bf8e642ca556e#diff-030fc115adbcd287aecbd623b8355c9cf9aa1fd05ef07b4001b05ce927ec5e03

OK to remove the tests completely then?

Reply:
Yes!